pages/certificate/[tool].tsx

```diff
@@ -0,0 +1,167 @@
+import { Page } from "types/Page";
+import { useRouter } from "next/router";
+import axios from "axios";
+import useSWR from "swr";
+import { ToolLayout } from "@components";
+import { Box, Button, Flex, Image, useBreakpointValue } from "@chakra-ui/react";
+import { ModuleHeader } from "@components/moduleHeader";
+import { createCertificate } from "src/utils/certificates";
+import { Auth } from "aws-amplify";
+import { useState, useEffect } from "react";
+
+const fetcher = async (url: string) => {
+    const response = await axios({
+        method: "GET",
+        url,
+    });
+    return response.data;
+};
+
+const CertificatePage: Page = () => {
+    const large = useBreakpointValue({ lg: true });
+    const router = useRouter();
+    const { tool } = router.query;
+    const [user, setUser] = useState<any | null>(null);
+    const fetchURL = tool ? `/api/tool/${tool}` : null;
+    const { data, error } = useSWR(() => fetchURL, fetcher);
+
+    const handleDownload = () => {
+        createCertificate(
+            user?.attributes?.name || "",
+            data?.title || "HeadsUpGuys Tool",
+        );
+    };
+    const handleShare = () => {
+        createCertificate(
+            user?.attributes?.name || "",
+            data?.title || "HeadsUpGuys Tool",
+        );
+    };
+    const handleBack = () => {
+        router.push("/");
+    };
+
+    useEffect(() => {
+        const getCurrentUser = async () => {
+            const currentUser = await Auth.currentAuthenticatedUser();
+            setUser(currentUser);
```

Comment:
Wonder if `setUser(await Auth.currentAuthenticatedUser())` would work? Not a big difference tho haha